src/Features/CSharp/Portable/InternalUtilities/CSharpOperationAnalysisHelpers.cs

```diff
@@ -0,0 +1,941 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Features
```

Comment:
I'm not sure this is the best place to put this. My thinking was that these operations are all static and could concieveably be used as helpers elsewhere. It's currently in "InternalUtilities", but a lot of these methods are fairly specific to the Convert To Record operation, and I'm not sure if any other features would actually get any use out of them, so I might move it back into the Convert To Record folder/namespace.

Reply:
Generally I'd say unless we're able to share the code already with another feature that's using it, just keep it in the convert to record code. If/when we need to share it, then we can refactor it accordingly.

Reply:
@CyrusNajmabadi might know too whether we've already got other code doing similar things. My memory quickly gets hazy on this stuff.

Reply:
When I talked to him about the feature in general, I think he was saying we don't have any of this general functionality except for the hash code analyzer. Some preliminary searches also didn't really bring anything up, but there might be some intermediate helper methods I could be using.